### ISSUE TITLE
Move zk-SNARK parameter path to shared app data location

### ISSUE THREAD

**Issue by nathan-at-least** [zcash/zcash#621]
In the alpha branch, zksnark parameters are placed into `~/.zcash-params`. But this is cutting corners. These values are a kind of "static application data" except we download them separately from our source repository.  If we had no concern about our code size, these could be simply bundled into our source repo and distributed with binary downloads.

So, they should live on the filesystem in a place for system-wide static application data, such as somewhere under `/usr/share` on linux as per http://www.pathname.com/fhs/pub/fhs-2.3.html#USRSHAREARCHITECTUREINDEPENDENTDATA

However, having giant binaries is cumbersome and we may wish to fetch these parameters in a similar manner to the rest of the blockchain, in which case they should live in a variable location such as somewhere under `/var/lib/`.

Also, of course, find the appropriate paths for each platform.

**Comment by jernst** [comment]
Suggest making this configurable. Could simply be an entry in the config file, with the default being the current implementation.

**Comment by daira** [comment]
IMHO the location should be /usr/share/zcash/params, and for simplicity, it should not be configurable. Also zcashd should check the hashes of the proving/verification keys on startup.

**Comment by nuttycom** [comment]
Related to #1312

**Comment by daira** [comment]
It's important that it remains possible to fully install zcashd as an unprivileged user. So if we add parameters in a shared location that requires root privileges to write to, zcashd must still be able to find them either in that location or the current one under the user's home directory.

**Comment by sellout** [comment]
There is a separate issue (#5503) for making the location configurable. Both are blocked on integrating fetch-params into zcashd rather than a separate process.

This issue is for adding a _standard_ cascading default. Roughly,
```bash
IFS=: read -r -d '' -a data_dirs < <(printf '%s:\0' "${XDG_DATA_DIRS}")
for data_dir in ${data_dirs[@]} + ("${XDG_DATA_HOME}"); do
  can_read_or_write "${data_dir}/zcash/zk-params"
done
```
although this thankfully won’t be in Bash at that point.

Writing to a shared location does imply that we’d still want to be able to explicitly fetch without running an instance (since we don’t want the zcashd running as root in general), but I think something like `sudo zcashd --fetch-params` is still better than a separate script.

**Comment by daira** [comment]
This would be mostly obviated by #3933 / #6661. For the remaining optional use of `sprout-groth16.params` there is not sufficient justification to implement this issue.
